experiment/simulation/js/main.js

```diff
@@ -1 +1,258 @@
-//Your JavaScript goes in here
+let detergentAddCounter = 1;
+
+// disable onclick of all components except of spoon and detergent
+if(document.getElementsByClassName("clothWithStain")[0]){
+    document.getElementsByClassName("clothWithStain")[0].style.pointerEvents = 'none';
+}
+if(document.getElementsByClassName("glassRod")[0]){
+    document.getElementsByClassName("glassRod")[0].style.pointerEvents = 'none';
+}
+if(document.getElementsByClassName("spoon2")[0]){
+    document.getElementsByClassName("spoon2")[0].style.pointerEvents = 'none';
+}
+if(document.getElementsByClassName("detergent2")[0]){
+    document.getElementsByClassName("detergent2")[0].style.pointerEvents = 'none';
+}
+if(document.getElementsByClassName("spoon3")[0]){
+    document.getElementsByClassName("spoon3")[0].style.pointerEvents = 'none';
+}
+if(document.getElementsByClassName("detergent3")[0]){
+    document.getElementsByClassName("detergent3")[0].style.pointerEvents = 'none';
+}
+
+
+// Functions for structure of soap part.
+// below function shows information about micelle parts when hovered upon
+function showMicelleInformation(part){
+    'use strict';
+    let tailInfo = document.getElementsByClassName("micelle-information")[0];
+    let info = "";
+    if(part === 'tail'){
+        info = "Hydrocarbon Chain(C<sub>7</sub>H<sub>35</sub>) </br> </br> Hydrophobic </br> (Water Repelling)";
+    }else if(part === 'head'){
+        info = "Ionic End(COO<sup>-</sup>Na<sup>+</sup>) </br> </br> Hydrophilic </br> (Water Attracting)";
+    }
+    tailInfo.innerHTML = info;
+}
+// The below functions hide the information about the micelle parts 
+// when cursor is not above them.
+function hideTailInformation(){
+    'use strict';
+    let tailInfo = document.getElementsByClassName("micelle-information")[0];
+    tailInfo.innerHTML = "";
+}
+
+function hideHeadInformation(){
+    'use strict';
+    let headInfo = document.getElementsByClassName("micelle-information")[0];
+    headInfo.innerHTML = "";
+}
+
+// Functions of Cleansing action
+// Below function moves the spoon and detergent from 
+// top right corner towards the beaker.
+function moveSpoon(){
+    'use strict';
+    let procedureMessage = document.getElementsByClassName("procedureMessage")[0]
+    let SCREENWIDTH = window.innerWidth
+    let changeYForDetergent = '24vw';
+    if(SCREENWIDTH > 1200){
+        changeYForDetergent = '19vw';
+    }
+
+    // disable onclick of spoon and detergent
+    document.getElementsByClassName("spoon"+detergentAddCounter)[0].style.pointerEvents = 'none';
+    document.getElementsByClassName("detergent"+detergentAddCounter)[0].style.pointerEvents = 'none';
```

Comment:
If you are looking for a specific element, it is better to get it byId rather than getting an array byClassName and then taking the first element. Think about a scenario where someone enhances the experiment to add another element and wants to reuse the classes for styling